android/java/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxgl/views/MapView.java

```diff
@@ -160,6 +160,30 @@
     // Used to manage Event Listeners
     private ArrayList<OnMapChangedListener> mOnMapChangedListener;
 
+    private enum MapChange {
+        MapChangeNullChange(-1),
+        MapChangeRegionWillChange(0),
+        MapChangeRegionWillChangeAnimated(1),
+        MapChangeRegionIsChanging(2),
+        MapChangeRegionDidChange(3),
+        MapChangeRegionDidChangeAnimated(4),
+        MapChangeWillStartLoadingMap(5),
+        MapChangeDidFinishLoadingMap(6),
+        MapChangeDidFailLoadingMap(7),
+        MapChangeWillStartRenderingFrame(8),
+        MapChangeDidFinishRenderingFrame(9),
+        MapChangeDidFinishRenderingFrameFullyRendered(10),
+        MapChangeWillStartRenderingMap(11),
+        MapChangeDidFinishRenderingMap(12),
+        MapChangeDidFinishRenderingMapFullyRendered(13);
+
+        private int value;
+
+        private MapChange(int value) {
+            this.value = value;
+        }
+    }
+
```

Comment:
This will definitely work, but if we don't need to maintain value for any particular reason it can definitely be safely removed.

```java
private enum MapChange {
MapChangeNullChange,
MapChangeRegionWillChange,
MapChangeRegionWillChangeAnimated,
...
}

// Comparison works this way then
MapChange mcEvent;
switch (mcEvent) {
   case MapChange.MapChangeNullChange:
      break;
   case MapChange.MapChangeRegionWillChange:
     break;
   case MapChange.MapChangeRegionWillChangeAnimated
     break;

....
   default:
     // Shouldn't get here
}

```

It can lead to a bit more verbose switch statements, but also safer ones as it doesn't require values to be copied and stored multiple places.